src/codegen_print.cpp

```
@@ -8,573 +7,19 @@ void CodeGen::emitPrint(const std::vector<ExprPtr> &args) {
 
     auto printfFn = getBufferedPrintf();
+    llvm::Constant *fmtS = cachedGlobalString("%s", ".fmt_print_s");
     llvm::Constant *space = args.size() > 1
         ? cachedGlobalString(" ", ".fmt_space") : nullptr;
 
     for (size_t i = 0; i < args.size(); ++i) {
         if (i > 0)
             builder_.CreateCall(printfFn, {space});
-        llvm::Value *val = emitExpr(*args[i]);
-        emitPrintSingle(val, printfFn);
+        llvm::Value *str = valueToString(emitExpr(*args[i]));
```

Cppcheck (2.20.0)

[information] Include file

(missingIncludeSystem)